tests/components/zha/test_light.py

```diff
@@ -449,119 +474,180 @@ async def async_test_zha_group_light_entity(
-    await dev1_cluster_on_off.on()
-    await dev2_cluster_on_off.on()
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 1})
+    await send_attributes_report(hass, dev2_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
 
     # test that group light is on
-    assert hass.states.get(entity_id).state == STATE_ON
+    assert hass.states.get(device_1_entity_id).state == STATE_ON
+    assert hass.states.get(device_2_entity_id).state == STATE_ON
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
-    await dev1_cluster_on_off.off()
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 0})
+    await hass.async_block_till_done()
 
     # test that group light is still on
-    assert hass.states.get(entity_id).state == STATE_ON
+    assert hass.states.get(device_1_entity_id).state == STATE_OFF
+    assert hass.states.get(device_2_entity_id).state == STATE_ON
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
-    await dev2_cluster_on_off.off()
+    await send_attributes_report(hass, dev2_cluster_on_off, {0: 0})
+    await hass.async_block_till_done()
 
     # test that group light is now off
-    assert hass.states.get(entity_id).state == STATE_OFF
+    assert hass.states.get(device_1_entity_id).state == STATE_OFF
+    assert hass.states.get(device_2_entity_id).state == STATE_OFF
+    assert hass.states.get(group_entity_id).state == STATE_OFF
 
-    await dev1_cluster_on_off.on()
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
 
     # test that group light is now back on
-    assert hass.states.get(entity_id).state == STATE_ON
+    assert hass.states.get(device_1_entity_id).state == STATE_ON
+    assert hass.states.get(device_2_entity_id).state == STATE_OFF
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
-    # test that group light is now off
-    await group_cluster_on_off.off()
-    assert hass.states.get(entity_id).state == STATE_OFF
+    # turn it off to test a new member add being tracked
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 0})
+    await hass.async_block_till_done()
+    assert hass.states.get(device_1_entity_id).state == STATE_OFF
+    assert hass.states.get(device_2_entity_id).state == STATE_OFF
+    assert hass.states.get(group_entity_id).state == STATE_OFF
 
     # add a new member and test that his state is also tracked
-    await zha_group.async_add_members([device_light_3.ieee])
-    await dev3_cluster_on_off.on()
-    assert hass.states.get(entity_id).state == STATE_ON
+    await zha_group.async_add_members([GroupMember(device_light_3.ieee, 1)])
+    await send_attributes_report(hass, dev3_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
+    assert device_3_entity_id in zha_group.member_entity_ids
+    assert len(zha_group.members) == 3
+
+    assert hass.states.get(device_1_entity_id).state == STATE_OFF
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)